File: tests/Fixtures/backend/app/Enums/OrderStatus.php
<?php

namespace Shop\Enums;

enum OrderStatus: string
{
    case Pending = 'pending';
    case Paid = 'paid';
    case Shipped = 'shipped';
    case Cancelled = 'cancelled';

    public function isTerminal(): bool
    {
        return match ($this) {
            self::Shipped, self::Cancelled => true,
            self::Pending, self::Paid => false,
        };
    }

    public function label(): string
    {
        return match ($this) {
            self::Pending => 'Pending',
            self::Paid => 'Paid',
            self::Shipped => 'Shipped',
            self::Cancelled => 'Cancelled',
        };
    }
}

File: tests/Fixtures/python/shop/order_status.py
# An order's statuses kept as a class of string constants — a closed set nothing seals, so any string
# passes where a status belongs. Below it, the FIX: a `StrEnum`, the set a type of its own.
from enum import StrEnum


# @sin ConstantClassEnum
class OrderStatus:
    PENDING = "pending"
    PAID = "paid"
    SHIPPED = "shipped"


# @fixed ConstantClassEnum
class OrderState(StrEnum):
    PENDING = "pending"
    PAID = "paid"
    SHIPPED = "shipped"
